samples/CommandsSandbox/src/main/java/commands/CommandsSandbox.java

```diff
@@ -0,0 +1,675 @@
+        } else {
+            deviceArn = context.mqttClientArn;
+        }
+
+        try {
+            GetCommandRequest getCommandRequest = GetCommandRequest.builder().commandId(commandId).build();
+            GetCommandResponse getCommandResponse = context.controlPlaneClient.getCommand(getCommandRequest);
+
+            StartCommandExecutionRequest request = StartCommandExecutionRequest.builder()
+                    .commandArn(getCommandResponse.commandArn())
+                    .executionTimeoutSeconds(timeout)
+                    .targetArn(deviceArn)
+                    .build();
+
+            context.iotJobsDataPlaneClient.startCommandExecution(request);
+            Thread.sleep(1000);
+        } catch (Exception ex) {
+            handleOperationException("send-command", ex, context);
+        }
+    }
+
+    private static void handleGetCommandExecution(ApplicationContext context, String arguments) {
+        String commandExecutionId = arguments.trim();
+
+        try {
+            if (context.activeCommandExecutions.containsKey(commandExecutionId)) {
+                CommandExecutionContext commandExecutionContext = context.activeCommandExecutions.get(commandExecutionId);
+
+                GetCommandExecutionRequest getCommandExecutionRequest = GetCommandExecutionRequest.builder()
+                        .executionId(commandExecutionId)
+                        .targetArn(commandExecutionContext.deviceArn)
+                        .build();
+                GetCommandExecutionResponse getCommandExecutionResponse = context.controlPlaneClient.getCommandExecution(getCommandExecutionRequest);
+                System.out.printf("Status of command execution '%s' is %s\n", commandExecutionId, getCommandExecutionResponse.status());
+                if (getCommandExecutionResponse.statusReason() != null) {
+                    System.out.printf("  Reason code: %s\n", getCommandExecutionResponse.statusReason().reasonCode());
+                    System.out.printf("  Reason description: %s\n", getCommandExecutionResponse.statusReason().reasonDescription());
+                }
+            }
+        } catch (Exception ex) {
+            handleOperationException("get-command-execution", ex, context);
+        }
+    }
+
+    private static void handleUpdateCommandExecution(ApplicationContext context, String arguments) {
+        String[] argumentSplit = arguments.trim().split(" ", 4);
+        if (argumentSplit.length < 2) {
+            printCommandHelp();
+            return;
+        }
+
+        String executionId = argumentSplit[0];
+        String statusStr = argumentSplit[1];
+
+        String reasonCode = null;
+        String reasonDescription = null;
+        if (argumentSplit.length > 3) {
+            reasonCode = argumentSplit[2];
+            reasonDescription = argumentSplit[3];
+        }
+
+        try {
+            CommandExecutionContext commandExecutionContext = context.activeCommandExecutions.get(executionId);
```

Comment:
Above you do a if (context.activeCommandExecutions.containsKey(commandExecutionId)) check but not here. Should we add one here to make sure we aren't trying to get a null? In the above `handleGetCommandExecution` it appears as though we don't provide any feedback at all on a non-existent commandExecutionID. In both cases, maybe we should notify that it doesn't exist?